genesis/utils/misc.py

```diff
@@ -323,19 +321,24 @@ def is_approx_multiple(a, b, tol=1e-7):
     return abs(a % b) < tol or abs(b - (a % b)) < tol
 
 
-def concat_with_tensor(tensor: torch.Tensor, value, expand: tuple[int, ...] | None = None, dim: int = 0):
+def concat_with_tensor(
+    tensor: torch.Tensor, value, expand: tuple[int, ...] | None = None, dim: int = 0, flatten: bool = False
+):
     """Helper method to concatenate a value (not necessarily a tensor) with a tensor."""
     if not isinstance(value, torch.Tensor):
         value = torch.tensor([value], dtype=tensor.dtype, device=tensor.device)
     if expand is not None:
         value = value.expand(*expand)
     if dim < 0:
         dim = tensor.ndim + dim
-    assert (
-        0 <= dim < tensor.ndim
-        and tensor.ndim == value.ndim
-        and all(e_1 == e_2 for i, (e_1, e_2) in enumerate(zip(tensor.shape, value.shape)) if e_1 > 0 and i != dim)
-    )
+    if flatten:
+        value = value.flatten()
+    else:
+        assert (
+            0 <= dim < tensor.ndim
+            and tensor.ndim == value.ndim
+            and all(e_1 == e_2 for i, (e_1, e_2) in enumerate(zip(tensor.shape, value.shape)) if e_1 > 0 and i != dim)
+        )
```

Comment:
This is wrong, you should flatten THEN assert, not skip the assert. The assert is here to check that `torch.cat` will have the expected behaviour, you cannot just skip it like this. As a rule of thumbs, never skip assert unless you really know what you are doing. Someone took the time to write them, it is for a reason.